### ISSUE TITLE
Loss function returns 0 after first epoch for training set only when using validation data in training

### ISSUE THREAD

**Issue by sabre-sr** [fabiodimarco/tf-levenberg-marquardt#18]
Hello and thank you for releasing your code.

I'm having an issue, where if I introduce a validation set into the training process, training loss value after first epoch is replaced by 0 in the output, as well as in History object - however validation loss is being calculated correctly, and is going down, indicating that the training is progressing despite loss being calculated as 0.

Example console output:
```
Epoch 1/760
8/8 [==============================] - 5s 187ms/step - damping_factor: 1.0000e-10 - attempts: 1.0000 - loss: 2.8319e-06 - r_square: -130.5820 - val_loss: 2.5453e-06 - val_r_square: -106.8921

Epoch 2/760
8/8 [==============================] - 1s 112ms/step - damping_factor: 1.0000e-10 - attempts: 1.0000 - loss: 0.0000e+00 - r_square: -106.5525 - val_loss: 2.0733e-06 - val_r_square: -86.5898

Epoch 3/760
8/8 [==============================] - 1s 134ms/step - damping_factor: 1.0000e-10 - attempts: 1.0000 - loss: 0.0000e+00 - r_square: -86.2367 - val_loss: 1.6919e-06 - val_r_square: -71.4224
```

Example code
```python
# X_train - DataFrame object
# Y_train - DataFrame object
model = Sequential()
model.add(Dense(34, input_shape=(48, ), activation='tanh'))
model.add(Dense(24, activation='linear'))
model_wrapper = lm.ModelWrapper(model)
model_wrapper.compile(
            optimizer=tf.keras.optimizers.SGD(learning_rate=0.01),
            loss=lm.MeanSquaredError(),
            metrics=RSquare())
early_stop = EarlyStopping(monitor='val_loss', patience=15)
history = model_wrapper.fit(X_train, y_train, epochs=760,
            validation_split=0.15,     # removing this parameter fixes the loss display
            callbacks=[early_stop])
```

**Comment by fabiodimarco** [comment]
Hi @sabre-sr, thank you for reporting the issue. I'll look into it later today. What are the input output dimensions of the data you are using?

**Comment by sabre-sr** [comment]
Thank you for your swift reply. 
My data has dimensions of 48 floats as inputs and 24 floats as outputs, with 365 samples - 48x365 and 24x365 respectively.
Modifications to sample size (48x760/24x760), as well as modifying validation split ratio or splitting data with train_test_split() from `sklearn` library didn't resolve the issue.

I'm also attaching library versions below:
```
Tensorflow/Keras: 2.10.0
pandas: 1.5.3
numpy: 1.24.2
sklearn: 1.2.1
```

Thank you in advance for your help.

**Comment by fabiodimarco** [comment]
I committed a fix for that. Let me know if it does work on your side.

**Comment by sabre-sr** [comment]
The fix works, thank you very much once again. The issue can be now closed.
